core/src/cost_update_service.rs

```diff
@@ -127,11 +130,10 @@ impl CostUpdateService {
                 continue;
             }
 
-            let units = program_timings.accumulated_units / program_timings.count as u64;
             cost_model
                 .write()
                 .unwrap()
-                .upsert_instruction_cost(program_id, units);
+                .upsert_instruction_cost(program_id, program_timings);
```

Comment:
passing `ProgramTiming` down to pipeline, so its detail can be reported when calculation anomaly is detected.